app/models/user.rb

```diff
@@ -10,6 +10,9 @@ class User < ActiveRecord::Base
                                                       inverse_of: :user, dependent: :destroy
   has_many :instance_users
   has_many :instances, through: :instance_users
+  has_many :activities, inverse_of: :actor, dependent: :destroy
+  has_many :notifications, through: :activities, inverse_of: :user, dependent: :destroy,
```

Comment:
I don't think you mean this. This goes through the activities relation. You want the notifications which target a given user.